File: appendix-a-frameworks/nextjs/lib/embedder.ts
import OpenAI from "openai";

const client = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });
const MODEL  = "text-embedding-3-small";

export async function embed(text: string): Promise<number[]> {
  const resp = await client.embeddings.create({ model: MODEL, input: text });
  return resp.data[0].embedding;
}

File: appendix-a-frameworks/django/rag/embedder.py
import os
from functools import lru_cache

from openai import OpenAI

_MODEL = "text-embedding-3-small"


@lru_cache(maxsize=1)
def _client() -> OpenAI:
    return OpenAI(api_key=os.environ["OPENAI_API_KEY"])


def embed(text: str) -> list[float]:
    resp = _client().embeddings.create(model=_MODEL, input=text)
    return resp.data[0].embedding
